### ISSUE TITLE
bug: snapcraft prints insufficient logging when using lxd

### ISSUE THREAD

**Issue by popey** [canonical/snapcraft#4933]
### Bug Description

Using LXD or a GitHub action such as [snapcore/action-build](https://github.com/snapcore/action-build/) (which uses LXD under the covers), snapcraft will often print up to, but excluding the most useful error message required for debugging. This is quite frustrating as the error may be a simple typo, or error in migration from core18 to core22 (where files have moved, or rules are more strict).

When using LXD locally, one can `snapcraft --debug` to jump into the container to view more details.

This isn't as straightforward when using LXD as part of a GitHub action run.

To find the error, I added the following conditional step after the standard entry for snapcore/action-build

```yaml
    steps:
    - uses: actions/checkout@v4
    - uses: snapcore/action-build@v1
      id: snapcraft
    - name: Build failed show log
      if: ${{ failure() }}
      run: cat /home/runner/.local/state/snapcraft/log/snapcraft*.log
    - uses: actions/upload-artifact@v2
      with:
        name: 'snap'
        path: ${{ steps.snapcraft.outputs.snap}}
```

This allowed me to uncover the single line that enabled me to debug the build:-

```
2024-07-22T19:55:52.2549267Z 2024-07-22 19:55:47.371 :: 2024-07-22 19:55:47.013 Detailed information: 2024/07/22 19:55:47.012142 container.go:393: in snap "twinejs": path "launch" does not exist
```

Should:

a) The action cat out the full log on failure?
b) snapcraft write out a more useful error on failure?

While a) is possible and likely simple to implement, b) would benefit GHA users, and local LXD users.

### To Reproduce

Create a yaml that has an error in the path to app command. This will result in an error during 'snap pack' - right at the end of the build. However, the error will not be printed but hidden inside a log file that you can't get to (when it's in a GitHub Action).

I put a hello-world yaml which *should* reproduce the issue, as I intentionally modified the command: line to be incorrect. In theory if you `snapcraft --use-lxd` it will build all the way up to `snap pack` then fail and not tell you why. 

### Environment

GitHub Action snapcore/action-build that uses lxd internally.

### snapcraft.yaml

```shell
name: hello
base: core22
version: '2.10'
summary: GNU Hello, the "hello world" snap
description: |
  GNU hello prints a friendly greeting.
grade: devel
confinement: devmode

apps:
  hello:
    command: hello

parts:
  gnu-hello:
    source: http://ftp.gnu.org/gnu/hello/hello-2.10.tar.gz
    plugin: autotools
```

### Relevant log output

```shell
Creating snap package...
Command '['snap', 'pack', '--filename', 'twinejs_2.9.1_amd64.snap', '--compression', 'lzo', PosixPath('/root/prime'), PosixPath('/root/project')]' returned non-zero exit status 1.
Failed to execute pack in instance.
Recommended resolution: Run the same command again with --debug to shell into the environment if you wish to introspect this failure.
Full execution log: '/home/runner/.local/state/snapcraft/log/snapcraft-20240722-190027.468199.log'
Error: The process '/usr/bin/sudo' failed with exit code 1
```

**Comment by mr-cal** [comment]
Hi @popey, thanks for the report!

We have had a few reports of this problem.  Right now, we're blocked by snapd needing to fix what `snap pack` returns.

I'm going to close this as a duplicate of #4499, where I'm tracking all details about this problem.
